java/events_slack_notifications/src/main/java/com/my_asana/app/MilestoneCelebratorySlackMessage.java

```diff
@@ -0,0 +1,141 @@
+package com.my_asana.app;
+
+import com.asana.Client;
+import com.asana.models.Event;
+import com.asana.models.Project;
+import com.asana.models.Workspace;
+import com.asana.models.Story;
+import com.asana.models.ResultBodyCollection;
+import com.asana.models.Task;
+import com.asana.errors.InvalidTokenError;
+
+import com.hubspot.slack.client.SlackClient;
+import com.hubspot.slack.client.SlackClientFactory;
+import com.hubspot.slack.client.SlackClientRuntimeConfig;
+import com.hubspot.slack.client.methods.params.chat.ChatPostMessageParams;
+import com.hubspot.slack.client.models.response.SlackError;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileInputStream;
+import java.util.List;
+
+public class MilestoneCelebratorySlackMessage {
+
+    private static String CHANNEL_ID = "#asana-notifications";
+
+    public static void main(String[] args) throws Exception {
+
+        Client client = getAsanaClient();
+        // SlackClient slackClient = getSlackClient();
+
+        String projectId = args[0];
+
+        // Attempt to get latest sync token.
+        File syncFile = new File("asana_sync_token_for_project_" + projectId);
+
+        // If the file does not exist, we fetch a new sync token for future use.
+        if (syncFile.createNewFile()) {
+            String initialSync = null;
+
+            try {
+                ResultBodyCollection<Event> eventResponse = client.events.get(projectId, "random_sync_token").executeRaw();
+            } catch(InvalidTokenError e) {
+                initialSync = e.sync;
+            }
+
+            writeSyncToken(syncFile, initialSync);
+            System.out.println("First run, saved sync token. Ready for future use.");
+            return;
+        }
+
+        String sync = readSyncToken(syncFile);
```

Comment:
I would pull out the `String sync;` to before the preceding if statement and have an else that reads. I'd prefer to only read if we don't have the information (if we write, no need to read cuz I think it's pretty slow).